### ISSUE TITLE
IPU slower than GPU

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#160]
Hey Graphcore team. I have ran some benchmarks comparing the performance of the IPU vs GPU on the QM9 dataset. The results are [available here](https://docs.google.com/spreadsheets/d/1GrDNK9Wd3gkthxrutfipuLixZh-X4gLxAMvngI1dcmo/edit?usp=sharing).
For now, I compare a single IPU vs the NVidia A6000 (48Gb), about 30-40% slower than the top A100 chip.
The results for IPU are overall slower. Twice slower for Validation, and a bit less for training. On both hardware, I have optimized the batch size to match the device requirements, but kept every other parameter the same. Everything runs on the branch gpu in Goli, and with precision 32 (no mixed precision).

GPU seems to benefit a lot from large batch sizes.

IPU suffers from delay from compilation (~10 min) and from delays at the start of each train epoch and validation epoch to reload the right model on the machine. These were ignored from the calculations both for IPU and GPUs. In the best case, IPUs perform at 4 seconds for the val epoch and 15 for the train (when ignoring delays), compared to 2/10 for GPUs. Also note that the GPU computes metrics on the training set, so it should be slower, technically.
Also, IPUs seem to suffer from larger models. When trying to fit a model of 42M of parameters, instead of 27M, reducing the batch size to ~10 could not fit on memory, compared to bz=50 with 27M parameter. This is compared to the 15,000 molecules that could fit on GPU for the 27M model.

To try to mimic the batch size = 6,000 from GPUs, I tried using gradient accumulation over 375 epochs with bz=20, but this slows down (just a bit) the computation and increases the memory consumption.
I used deviceIterations=4.

In summary, there seems to still be work needed to match, and eventually surpass, the performance of GPUs. I don't have the pop-vision profile to share. I'll generate one and share it tomorrow.

What's your experience with Tensorflow? Do you have faster computation than Gpus? Do you still expect a 2-3x improvement that was pitched a few months earlier?

In my experiments, the time between the steps is low, meaning that the `deviceIterations` is not the bottleneck. See this image
![image](https://user-images.githubusercontent.com/47570400/200849441-85ba4c7d-0fbc-43e4-9d44-f5dc379f5679.png)

However, most of the IPU time is spent waiting with `IPUSync::waitMarkCountIsLessEqualThan`, see this image 
![image](https://user-images.githubusercontent.com/47570400/200849785-3f6f9512-1ab7-4046-98e0-41addb9ba855.png)
